internal/testrunner/runners/pipeline/runner.go

```diff
@@ -0,0 +1,158 @@
+package pipeline
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/elasticsearch"
+	"github.com/elastic/elastic-package/internal/packages"
+	"github.com/elastic/elastic-package/internal/testrunner"
+)
+
+const (
+	// TestType defining pipeline tests
+	TestType testrunner.TestType = "pipeline"
+)
+
+type runner struct {
+	options testrunner.TestOptions
+}
+
+// Run runs the pipeline tests defined under the given folder
+func Run(options testrunner.TestOptions) error {
+	r := runner{options}
+	return r.run()
+}
+
+func (r *runner) run() error {
+	testCaseFiles, err := r.listTestCaseFiles()
+	if err != nil {
+		return errors.Wrap(err, "listing test case definitions failed")
+	}
+
+	datasetPath, found, err := packages.FindDatasetRootForPath(r.options.TestFolderPath)
+	if err != nil {
+		return errors.Wrap(err, "locating dataset root failed")
+	}
+	if !found {
+		return errors.New("dataset root not found")
+	}
+
+	esClient, err := elasticsearch.Client()
+	if err != nil {
+		return errors.Wrap(err, "fetching Elasticsearch client instance failed")
+	}
+
+	entryPipeline, pipelineIDs, err := installIngestPipelines(esClient, datasetPath)
+	if err != nil {
+		return errors.Wrap(err, "installing ingest pipelines failed")
+	}
+	defer func() {
+		err := uninstallIngestPipelines(esClient, pipelineIDs)
+		if err != nil {
+			fmt.Printf("uninstalling ingest pipelines failed: %v", err)
```

Comment:
Should this use the logger?

Reply:
Based on our discussion in https://github.com/elastic/elastic-package/pull/73/files#r475905214, I would consider this to be a logging statement, not part of the output of the command.

Reply:
Fixed